### ISSUE TITLE
Generic callback protocols with `ParamSpec` are not inferring arguments correctly

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2433]
**Describe the bug**
Given the following code:

```python
from typing import Generic, Protocol
from typing_extensions import ParamSpec

P = ParamSpec('P')


class Callback(Protocol[P]):
    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> None:
        ...


def invoke(cb: Callback[P], /, *args: P.args, **kwargs: P.kwargs) -> None:
    cb(*args, **kwargs)


def compare(a: int, b: int) -> None:
    pass


invoke(compare, 1, 2)  # Error: expected 0 positional arguments


class Command(Generic[P]):
    def __init__(self, callback: Callback[P]) -> None:
        self.callback: Callback[P] = callback

    def invoke(self, *args: P.args, **kwargs: P.kwargs) -> None:
        self.callback(*args, **kwargs)


command = Command(compare)
command.invoke(1, 2)  # Error: expected 0 positional arguments
reveal_type(command)  # Type of "command" is "Command[()]"
```

Errors are generated at indicated lines above and the type of `command` is reported with an incorrectly inferred argument list.

**Expected behavior**
No errors should be reported and the type of `command` should be `Command[(a: int, b: int)]`

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.178

**Comment by erictraut** [comment]
Thanks for the bug report. I agree that it's a bug in the type evaluator. I'll investigate further.

**Comment by erictraut** [comment]
There were two separate bugs here. The one that was causing almost all of the false positives in your samples above was a simple off-by-one error that eluded my unit tests — embarrassing, but thankfully an easy fix. https://github.com/microsoft/pyright/commit/5561df7a2aad8ba39f2178ea158d2906148ba79f

Both bugs will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.179, which I just published. It will also be included in the next release of pylance.
